### ISSUE TITLE
Issue with environments with scalar action: Action is not atleast 1D

### ISSUE THREAD

**Issue by simon-bachhuber** [aai-institute/tianshou#607]
I am not quite sure if this is a bug or not. 
The issue is that the environment expects an action of shape `action.shape=(1,)` but somehow the environment ends up receiving `action.shape=(,)`.

Code to replicate 
```python
import gym
import numpy as np
import torch

from tianshou.data import Collector, ReplayBuffer
from tianshou.policy import PPOPolicy
from tianshou.trainer import onpolicy_trainer
from tianshou.utils.net.common import ActorCritic, Net
from tianshou.utils.net.continuous import Actor, Critic
import dmc2gym

import warnings
warnings.filterwarnings('ignore')

device = 'cuda' if torch.cuda.is_available() else 'cpu'

# The error is raised for both envs at different times but the cause is always a missing dimension
#env_fn = lambda: dmc2gym.make("cartpole", "swingup")
env_fn = lambda: gym.make("MountainCarContinuous-v0")
env = env_fn()

# net is the shared head of the actor and the critic
net = Net(env.observation_space.shape, hidden_sizes=[64, 64], device=device)
actor = Actor(net, env.action_space.shape, device=device).to(device)
critic = Critic(net, device=device).to(device)
actor_critic = ActorCritic(actor, critic)

# optimizer of the actor and the critic
optim = torch.optim.Adam(actor_critic.parameters(), lr=0.0003)

dist = lambda logits: torch.distributions.Normal(logits, 1)
policy = PPOPolicy(actor, critic, optim, dist, action_space=env.action_space, deterministic_eval=True)

train_collector = Collector(policy, env, ReplayBuffer(20000))
test_collector = Collector(policy, env_fn())

result = onpolicy_trainer(
    policy,
    train_collector,
    test_collector,
    max_epoch=10,
    step_per_epoch=50000,
    repeat_per_collect=10,
    episode_per_test=10,
    batch_size=256,
    step_per_collect=200,
    stop_fn=lambda mean_reward: mean_reward >= 195,
)
```

**Comment by Trinkle23897** [comment]
It's because you use the network for discrete action space to train with continuous action space env (MCC has a continuous action space, while cartpole is discrete). Please refer to https://github.com/thu-ml/tianshou/blob/master/test/continuous/test_ppo.py to see the correct usage.

The difference is: `Actor` generates logits for Categorical distribution, while `ActorProb` generates `(mu, sigma)` for Gaussian distribution. The above code only generates `mu` instead of both `mu` and `sigma`.

**Comment by simon-bachhuber** [comment]
The `dm_control.suite.cartpole.swingup` has also a continuous action space.

I don't really understand the second aspect you mentioned. I use
`from tianshou.utils.net.continuous import Actor, Critic` 
because the environments require a continuous action. I expected `Actor` to return some numerical output of shape `env.action_space.shape` these values then get passed into `dist` such that they represent the mean of a Normal distribution. From this distribution the policy can be sampled to obtain actions and the log-prob can be calculated.

Where did i take a wrong turn? :)

**Comment by Trinkle23897** [comment]
https://github.com/thu-ml/tianshou/blob/e01385ea30d21b7efc3f9377d22d83e6afd39e2f/test/continuous/test_ppo.py#L84

> dist = lambda logits: torch.distributions.Normal(logits, 1)

You use a fixed sigma, that's not the correct way.

https://github.com/thu-ml/tianshou/blob/e01385ea30d21b7efc3f9377d22d83e6afd39e2f/test/continuous/test_ppo.py#L99-L102

Here the logits is `(mu, sigma)` instead of only mu.

**Comment by simon-bachhuber** [comment]
Thank you.
First of all with your two suggestions **it works just fine**.

Regardless, i still fail to see why my first approach didn't even run. 
Doesn't work:
`Actor: obs -> logits`
`dist: logits -> action`
Works:
`ActorProb: obs -> mu,sigma`
`dist: mu,sigma -> action`

He must have somewhere "consumed" a dimension due to this and then the action ended up with one missing dimension.

Feel free to close this issue as you see fit.

**Comment by Trinkle23897** [comment]
It's because https://github.com/thu-ml/tianshou/blob/e01385ea30d21b7efc3f9377d22d83e6afd39e2f/tianshou/policy/modelfree/sac.py#L118-L121

At first, it runs with eval mode. So `act = logits[0]`. However, here we assume `logits` is a tuple of `mu, sigma` instead of a single tensor.
